FirebaseAuth/Sources/Swift/Utilities/AuthRecaptchaVerifier.swift

```diff
@@ -125,38 +125,75 @@
-        return try await withCheckedThrowingContinuation { continuation in
-          FIRRecaptchaGetToken(siteKey, actionString,
-                               "NO_RECAPTCHA") { (token: String, error: Error?,
-                                                  linked: Bool, actionCreated: Bool) in
-              guard linked else {
-                continuation.resume(throwing: AuthErrorUtils.recaptchaSDKNotLinkedError())
-                return
-              }
-              guard actionCreated else {
-                continuation.resume(throwing: AuthErrorUtils.recaptchaActionCreationFailed())
-                return
-              }
-              if let error {
-                continuation.resume(throwing: error)
-                return
-              } else {
-                if token == "NO_RECAPTCHA" {
-                  AuthLog.logInfo(code: "I-AUT000031",
-                                  message: "reCAPTCHA token retrieval failed. NO_RECAPTCHA sent as the fake code.")
-                } else {
-                  AuthLog.logInfo(
-                    code: "I-AUT000030",
-                    message: "reCAPTCHA token retrieval succeeded."
-                  )
-                }
-                continuation.resume(returning: token)
-              }
-          }
+
+        let (token, error, linked, actionCreated) = await recaptchaToken(
+          siteKey: siteKey,
+          actionString: actionString,
+          fakeToken: "NO_RECAPTCHA"
+        )
+
+        guard linked else {
+          throw AuthErrorUtils.recaptchaSDKNotLinkedError()
+        }
+        guard actionCreated else {
+          throw AuthErrorUtils.recaptchaActionCreationFailed()
+        }
+        if let error {
+          throw error
         }
+        if token == "NO_RECAPTCHA" {
+          AuthLog.logInfo(code: "I-AUT000031",
+                          message: "reCAPTCHA token retrieval failed. NO_RECAPTCHA sent as the fake code.")
+        } else {
+          AuthLog.logInfo(
+            code: "I-AUT000030",
+            message: "reCAPTCHA token retrieval succeeded."
+          )
+        }
+        return token
       #endif // !(COCOAPODS || SWIFT_PACKAGE)
     }
 
+    private static var recaptchaClient: (any RCARecaptchaClientProtocol)?
+
+    private func recaptchaToken(siteKey: String,
+                                actionString: String,
+                                fakeToken: String) async -> (token: String, error: Error?,
+                                                             linked: Bool, actionCreated: Bool) {
+      if recaptchaClient != nil {
+        return await retrieveToken(actionString: actionString, fakeToken: fakeToken)
+      }
+
+      if let recaptcha =
+        NSClassFromString("RecaptchaEnterprise.RCARecaptcha") as? RCARecaptchaProtocol.Type {
+        do {
+          // let client = try await recaptcha.fetchClient(withSiteKey: siteKey)
+          let client = try await recaptcha.getClient(withSiteKey: siteKey)
```

Comment:
when recaptcha SDK is released:
```suggestion
          let client = try await recaptcha.fetchClient(withSiteKey: siteKey)
```